endpoints/put.go

```diff
@@ -144,20 +144,28 @@ func parsePutObject(p putObject) (string, error) {
 	return toCache, nil
 }
 
-func logBackendError(err error, index int) error {
+func classifyBackendError(err error, index int) error {
 	if _, ok := err.(*backendDecorators.BadPayloadSize); ok {
 		return utils.NewPBCError(utils.BAD_PAYLOAD_SIZE, fmt.Sprintf("POST /cache element %d exceeded max size: %v", index, err.Error()))
 	}
 
-	logrus.Error("POST /cache Error while writing to the backend: ", err)
-
-	if err == context.DeadlineExceeded {
-		logrus.Error("POST /cache timed out:", err)
+	switch err {
+	case context.DeadlineExceeded:
 		return utils.NewPBCError(utils.PUT_DEADLINE_EXCEEDED)
+	default:
+		return utils.NewPBCError(utils.PUT_INTERNAL_SERVER, err.Error())
 	}
 
+	return nil
+}
+
+func logBackendError(err error) {
+	logrus.Error("POST /cache Error while writing to the back-end: ", err)
+
+	if pbcErr, isPBCErr := err.(utils.PBCError); isPBCErr && pbcErr.StatusCode == utils.PUT_DEADLINE_EXCEEDED {
+		logrus.Error("POST /cache timed out:", err)
+	}
 	logrus.Error("POST /cache had an unexpected error:", err)
```

Comment:
> Do you want to log both of these messages if the status code is deadline exceeded?


Good catch. No, is either one or the other. In the future we might want to revisit these log entries but for the moment I'm leaving things are they currently are in production:
**Production**
```
    if len(resps.Responses[i].UUID) > 0 {
        err = backend.Put(ctx, resps.Responses[i].UUID, toCache, p.TTLSeconds)
        if err != nil {
            if _, ok := err.(*backendDecorators.BadPayloadSize); ok {
                http.Error(w, fmt.Sprintf("POST /cache element %d exceeded max size: %v", i, err), http.StatusBadRequest)
                return
            }

            logrus.Error("POST /cache Error while writing to the backend: ", err)
            switch err {
            case context.DeadlineExceeded:
                logrus.Error("POST /cache timed out:", err)
                http.Error(w, "Timeout writing value to the backend", HttpDependencyTimeout)
            default:
                logrus.Error("POST /cache had an unexpected error:", err)
                http.Error(w, err.Error(), http.StatusInternalServerError)
            }
            return
        }
        logrus.Tracef("PUT /cache uuid=%s", resps.Responses[i].UUID)
    }
```
The `default` clause forces to either print one or the other so this was indeed an error. Corrected.
```
162   func logBackendError(err error) {
163       logrus.Error("POST /cache Error while writing to the back-end: ", err)
164  
165       if pbcErr, isPBCErr := err.(utils.PBCError); isPBCErr && pbcErr.StatusCode == utils.PUT_DEADLINE_EXCEEDED {
166           logrus.Error("POST /cache timed out:", err)
167 -     }
168 -     logrus.Error("POST /cache had an unexpected error:", err)
    +     } else {
    +         logrus.Error("POST /cache had an unexpected error:", err)
    +     }
169   }
endpoints/put.go
```

Reply:
> Also I assume you're logging /cache timed out along with the error because the error message doesn't provide enough info indicating a time out is the cause, is that correct?

Yes, the `err` param is passed to the `logrus.Error()` call and should print the error.